src/Buffer.h

```diff
@@ -78,6 +78,29 @@ std::vector<int> get_shape_from_start_of_parameter_pack(Args &&... args) {
 template<typename T, typename T2>
 using add_const_if_T_is_const = typename std::conditional<std::is_const<T>::value, const T2, T2>::type;
 
+// Helpers to produce the name of a Buffer element type (a Halide
+// scalar type, or void, possibly with const). Useful for an error
+// messages.
+template<typename T>
+void buffer_type_name_non_const(std::ostringstream &oss) {
```

Comment:
Nit: using `ostream` instead of `ostringstream` would be fine and more general

Reply:
Done.